### PR TITLE
SSE: Add Classic conditions editor

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds a new expression type. classic conditions which is a port of the conditions editor from the Alerting tab in panel edit. This is needed to enable migrations from panel alerts to unified alert rules.

**Special notes for your reviewer**: Go to `http://localhost:3000/alerting/ng/list` and click Add NG Alert. After that, add one or more queries then add an expression and choose classic conditions from the drop down.

### CODE REVIEW

File: public/app/features/expressions/types.ts [hunk 1]
```diff
@@ -1,20 +1,82 @@
-import { DataQuery } from '@grafana/data';
+import { DataQuery, ReducerID, SelectableValue } from '@grafana/data';
 
-export enum GELQueryType {
+export enum ExpressionQueryType {
   math = 'math',
   reduce = 'reduce',
   resample = 'resample',
+  classic = 'classic',
```
Comment (kylebrandt) [inline public/app/features/expressions/types.ts:7]:
```suggestion
  classic = 'classic_conditions',
```

**Comment by kylebrandt** [comment]
Seems to be generally working from a dashboard view at least:

 1. "HAS NO VALUE" should so no argument (that is how it behaves in dashboard alerting)
 2. When removing a condition in the editor via the Delete button, the `conditions` array includes a null entry, which is causing the backend to throw an error. Can that be fixed so there isn't a null entry in the array?

image

**Comment by kylebrandt** [comment]
Nice Progress, using the translation endpoint on this nonsense:

image

Produces what I would expect: 

image

Once the time range shows up things should match

**Comment by kylebrandt** [comment]
Mark as ready for review? I am hoping we can hide it when in the dashboard view, but can do that in followup PR later when we have more workable views.

**Review by kylebrandt — APPROVED** [review]
Works on my instance :-)

**Review by domasx2 — APPROVED** [review]
Looks good, don't have much to add :) will soon be at the stage where we can integrate with the new alert form, will reach out for help

**Pull request merged**